bug: mysql table with bad datetime data casues ibis to fail to connect to table

### What happened?

Given poorly define `date` column in a table, ibis fails to identify the column and does not want to read it at all. 

Sample table:
```
name  tradedate                date
   C 2018-10-22 0000-00-00 00:00:00
   B 2017-06-07 0000-00-00 00:00:00
   C 2022-12-21 0000-00-00 00:00:00
```
on a command to connect to such a table, `idb.table('ztmp_date_issue')`, ibis comes back with a loud error as shown in the log output below. It appears pandas is complaining. 

I would have liked that it tells me at least which column is having issue with and allow me to ignore such columns. It can be a chore to find such columns when there are many of them. 

A workaround, look at datetime columns in the database and find the one that may have incorrectly formatted dates and exclude it via select as shown here:  `idb.table('ztmp_date_issue').select( ~ s.c('date')`

Schema for the sample table:
```
ibis.Schema {
  name       string
  tradedate  timestamp
  date       timestamp
}

or in sql: 
create table ztmp_date_issue
(
    name      char(10) null,
    tradedate datetime                   not null,
    date      datetime                   null
);

```




### What version of ibis are you using?

5.1

### What backend(s) are you using, if any?

mysql

### Relevant log output

```sh
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
File ....envs\py310\lib\site-packages\dateutil\parser\_parser.py:649, in parser.parse(self, timestr, default, ignoretz, tzinfos, **kwargs)
    648 try:
--> 649     ret = self._build_naive(res, default)
    650 except ValueError as e:

File ....envs\py310\lib\site-packages\dateutil\parser\_parser.py:1235, in parser._build_naive(self, res, default)
   1233         repl['day'] = monthrange(cyear, cmonth)[1]
-> 1235 naive = default.replace(**repl)
   1237 if res.weekday is not None and not res.day:

ValueError: year 0 is out of range

The above exception was the direct cause of the following exception:

ParserError                               Traceback (most recent call last)
File ....envs\py310\lib\site-packages\pandas\_libs\tslib.pyx:605, in pandas._libs.tslib.array_to_datetime()

File ....envs\py310\lib\site-packages\pandas\_libs\tslibs\parsing.pyx:318, in pandas._libs.tslibs.parsing.parse_datetime_string()

File ....envs\py310\lib\site-packages\dateutil\parser\_parser.py:1368, in parse(timestr, parserinfo, **kwargs)
   1367 else:
-> 1368     return DEFAULTPARSER.parse(timestr, **kwargs)

File ....envs\py310\lib\site-packages\dateutil\parser\_parser.py:651, in parser.parse(self, timestr, default, ignoretz, tzinfos, **kwargs)
    650 except ValueError as e:
--> 651     six.raise_from(ParserError(str(e) + ": %s", timestr), e)
    653 if not ignoretz:

File :3, in raise_from(value, from_value)

ParserError: year 0 is out of range: 0000-00-00 00:00:00

During handling of the above exception, another exception occurred:

TypeError                                 Traceback (most recent call last)
File ....envs\py310\lib\site-packages\pandas\_libs\tslib.pyx:616, in pandas._libs.tslib.array_to_datetime()

TypeError: invalid string coercion to datetime for "0000-00-00 00:00:00" at position 0

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
File ....envs\py310\lib\site-packages\dateutil\parser\_parser.py:649, in parser.parse(self, timestr, default, ignoretz, tzinfos, **kwargs)
    648 try:
--> 649     ret = self._build_naive(res, default)
    650 except ValueError as e:

File ....envs\py310\lib\site-packages\dateutil\parser\_parser.py:1235, in parser._build_naive(self, res, default)
   1233         repl['day'] = monthrange(cyear, cmonth)[1]
-> 1235 naive = default.replace(**repl)
   1237 if res.weekday is not None and not res.day:

ValueError: year 0 is out of range

The above exception was the direct cause of the following exception:

ParserError                               Traceback (most recent call last)
File ....envs\py310\lib\site-packages\IPython\core\formatters.py:974, in MimeBundleFormatter.__call__(self, obj, include, exclude)
    971     method = get_real_method(obj, self.print_method)
    973     if method is not None:
--> 974         return method(include=include, exclude=exclude)
    975     return None
    976 else:

File ....envs\py310\lib\site-packages\ibis\expr\types\core.py:33, in _FixedTextJupyterMixin._repr_mimebundle_(self, *args, **kwargs)
     32 def _repr_mimebundle_(self, *args, **kwargs):
---> 33     bundle = super()._repr_mimebundle_(*args, **kwargs)
     34     bundle["text/plain"] = bundle["text/plain"].rstrip()
     35     return bundle

File ....envs\py310\lib\site-packages\rich\jupyter.py:48, in JupyterMixin._repr_mimebundle_(self, include, exclude, **kwargs)
     41 def _repr_mimebundle_(
     42     self: "ConsoleRenderable",
     43     include: Sequence[str],
     44     exclude: Sequence[str],
     45     **kwargs: Any,
     46 ) -> Dict[str, str]:
     47     console = get_console()
---> 48     segments = list(console.render(self, console.options))
     49     html = _render_segments(segments)
     50     text = console._render_buffer(segments)

File ....envs\py310\lib\site-packages\rich\console.py:1311, in Console.render(self, renderable, options)
   1309 renderable = rich_cast(renderable)
   1310 if hasattr(renderable, "__rich_console__") and not isclass(renderable):
-> 1311     render_iterable = renderable.__rich_console__(self, _options)  # type: ignore[union-attr]
   1312 elif isinstance(renderable, str):
   1313     text_renderable = self.render_str(
   1314         renderable, highlight=_options.highlight, markup=_options.markup
   1315     )

File ....envs\py310\lib\site-packages\ibis\expr\types\relations.py:260, in Table.__rich_console__(self, console, options)
    257 else:
    258     width = options.max_width
--> 260 table = to_rich_table(self, width)
    261 return console.render(table, options=options)

File ....envs\py310\lib\site-packages\ibis\expr\types\pretty.py:229, in to_rich_table(table, console_width)
    227 # Compute the data and return a pandas dataframe
    228 nrows = ibis.options.repr.interactive.max_rows
--> 229 result = table.limit(nrows + 1).execute()
    231 # Now format the columns in order, stopping if the console width would
    232 # be exceeded.
    233 col_info = []

File ....envs\py310\lib\site-packages\ibis\expr\types\core.py:303, in Expr.execute(self, limit, timecontext, params, **kwargs)
    276 def execute(
    277     self,
    278     limit: int | str | None = 'default',
   (...)
    281     **kwargs: Any,
    282 ):
    283     """Execute an expression against its backend if one exists.
    284 
    285     Parameters
   (...)
    301         Keyword arguments
    302     """
--> 303     return self._find_backend(use_default=True).execute(
    304         self, limit=limit, timecontext=timecontext, params=params, **kwargs
    305     )

File ....envs\py310\lib\site-packages\ibis\backends\base\sql\__init__.py:254, in BaseSQLBackend.execute(self, expr, params, limit, **kwargs)
    251 self._register_in_memory_tables(expr)
    253 with self._safe_raw_sql(sql, **kwargs) as cursor:
--> 254     result = self.fetch_from_cursor(cursor, schema)
    256 if hasattr(getattr(query_ast, 'dml', query_ast), 'result_handler'):
    257     result = query_ast.dml.result_handler(result)

File ....envs\py310\lib\site-packages\ibis\backends\base\sql\alchemy\__init__.py:158, in BaseAlchemyBackend.fetch_from_cursor(self, cursor, schema)
    155 import pandas as pd
    157 df = pd.DataFrame.from_records(cursor, columns=schema.names, coerce_float=True)
--> 158 df = schema.apply_to(df)
    159 if not df.empty and geospatial_supported:
    160     return self._to_geodataframe(df, schema)

File ....envs\py310\lib\site-packages\ibis\expr\schema.py:305, in Schema.apply_to(self, df)
    302     not_equal = True
    304 if not_equal or not dtype.is_primitive():
--> 305     new_col = convert(col_dtype, dtype, col)
    306 else:
    307     new_col = col

File ....envs\py310\lib\site-packages\multipledispatch\dispatcher.py:278, in Dispatcher.__call__(self, *args, **kwargs)
    276     self._cache[types] = func
    277 try:
--> 278     return func(*args, **kwargs)
    280 except MDNotImplementedError:
    281     funcs = self.dispatch_iter(*types)

File ....envs\py310\lib\site-packages\ibis\backends\pandas\client.py:180, in convert_any_to_timestamp(_, out_dtype, column)
    177 @sch.convert.register(np.dtype, dt.Timestamp, pd.Series)
    178 def convert_any_to_timestamp(_, out_dtype, column):
    179     try:
--> 180         return column.astype(out_dtype.to_pandas())
    181     except pd.errors.OutOfBoundsDatetime:
    182         try:

File ....envs\py310\lib\site-packages\pandas\core\generic.py:6240, in NDFrame.astype(self, dtype, copy, errors)
   6233     results = [
   6234         self.iloc[:, i].astype(dtype, copy=copy)
   6235         for i in range(len(self.columns))
   6236     ]
   6238 else:
   6239     # else, only a single dtype is given
-> 6240     new_data = self._mgr.astype(dtype=dtype, copy=copy, errors=errors)
   6241     return self._constructor(new_data).__finalize__(self, method="astype")
   6243 # GH 33113: handle empty frame or series

File ....envs\py310\lib\site-packages\pandas\core\internals\managers.py:448, in BaseBlockManager.astype(self, dtype, copy, errors)
    447 def astype(self: T, dtype, copy: bool = False, errors: str = "raise") -> T:
--> 448     return self.apply("astype", dtype=dtype, copy=copy, errors=errors)

File ....envs\py310\lib\site-packages\pandas\core\internals\managers.py:352, in BaseBlockManager.apply(self, f, align_keys, ignore_failures, **kwargs)
    350         applied = b.apply(f, **kwargs)
    351     else:
--> 352         applied = getattr(b, f)(**kwargs)
    353 except (TypeError, NotImplementedError):
    354     if not ignore_failures:

File ....envs\py310\lib\site-packages\pandas\core\internals\blocks.py:526, in Block.astype(self, dtype, copy, errors)
    508 """
    509 Coerce to the new dtype.
    510 
   (...)
    522 Block
    523 """
    524 values = self.values
--> 526 new_values = astype_array_safe(values, dtype, copy=copy, errors=errors)
    528 new_values = maybe_coerce_values(new_values)
    529 newb = self.make_block(new_values)

File ....envs\py310\lib\site-packages\pandas\core\dtypes\astype.py:299, in astype_array_safe(values, dtype, copy, errors)
    296     return values.copy()
    298 try:
--> 299     new_values = astype_array(values, dtype, copy=copy)
    300 except (ValueError, TypeError):
    301     # e.g. astype_nansafe can fail on object-dtype of strings
    302     #  trying to convert to float
    303     if errors == "ignore":

File ....envs\py310\lib\site-packages\pandas\core\dtypes\astype.py:230, in astype_array(values, dtype, copy)
    227     values = values.astype(dtype, copy=copy)
    229 else:
--> 230     values = astype_nansafe(values, dtype, copy=copy)
    232 # in pandas we don't store numpy str dtypes, so convert to object
    233 if isinstance(dtype, np.dtype) and issubclass(values.dtype.type, str):

File ....envs\py310\lib\site-packages\pandas\core\dtypes\astype.py:151, in astype_nansafe(arr, dtype, copy, skipna)
    147 if is_datetime64_dtype(dtype):
    148     from pandas import to_datetime
    150     return astype_nansafe(
--> 151         to_datetime(arr.ravel()).values.reshape(arr.shape),
    152         dtype,
    153         copy=copy,
    154     )
    155 elif is_timedelta64_dtype(dtype):
    156     # bc we know arr.dtype == object, this is equivalent to
    157     #  `np.asarray(to_timedelta(arr))`, but using a lower-level API that
    158     #  does not require a circular import.
    159     return array_to_timedelta64(arr).view("m8[ns]").astype(dtype, copy=False)

File ....envs\py310\lib\site-packages\pandas\core\tools\datetimes.py:1100, in to_datetime(arg, errors, dayfirst, yearfirst, utc, format, exact, unit, infer_datetime_format, origin, cache)
   1098         result = _convert_and_box_cache(argc, cache_array)
   1099     else:
-> 1100         result = convert_listlike(argc, format)
   1101 else:
   1102     result = convert_listlike(np.array([arg]), format)[0]

File ....envs\py310\lib\site-packages\pandas\core\tools\datetimes.py:438, in _convert_listlike_datetimes(arg, format, name, tz, unit, errors, infer_datetime_format, dayfirst, yearfirst, exact)
    436 assert format is None or infer_datetime_format
    437 utc = tz == "utc"
--> 438 result, tz_parsed = objects_to_datetime64ns(
    439     arg,
    440     dayfirst=dayfirst,
    441     yearfirst=yearfirst,
    442     utc=utc,
    443     errors=errors,
    444     require_iso8601=require_iso8601,
    445     allow_object=True,
    446 )
    448 if tz_parsed is not None:
    449     # We can take a shortcut since the datetime64 numpy array
    450     # is in UTC
    451     dta = DatetimeArray(result, dtype=tz_to_dtype(tz_parsed))

File ....envs\py310\lib\site-packages\pandas\core\arrays\datetimes.py:2177, in objects_to_datetime64ns(data, dayfirst, yearfirst, utc, errors, require_iso8601, allow_object, allow_mixed)
   2175 order: Literal["F", "C"] = "F" if flags.f_contiguous else "C"
   2176 try:
-> 2177     result, tz_parsed = tslib.array_to_datetime(
   2178         data.ravel("K"),
   2179         errors=errors,
   2180         utc=utc,
   2181         dayfirst=dayfirst,
   2182         yearfirst=yearfirst,
   2183         require_iso8601=require_iso8601,
   2184         allow_mixed=allow_mixed,
   2185     )
   2186     result = result.reshape(data.shape, order=order)
   2187 except OverflowError as err:
   2188     # Exception is raised when a part of date is greater than 32 bit signed int

File ....envs\py310\lib\site-packages\pandas\_libs\tslib.pyx:427, in pandas._libs.tslib.array_to_datetime()

File ....envs\py310\lib\site-packages\pandas\_libs\tslib.pyx:683, in pandas._libs.tslib.array_to_datetime()

File ....envs\py310\lib\site-packages\pandas\_libs\tslib.pyx:829, in pandas._libs.tslib._array_to_datetime_object()

File ....envs\py310\lib\site-packages\pandas\_libs\tslib.pyx:819, in pandas._libs.tslib._array_to_datetime_object()

File ....envs\py310\lib\site-packages\pandas\_libs\tslibs\parsing.pyx:318, in pandas._libs.tslibs.parsing.parse_datetime_string()

File ....envs\py310\lib\site-packages\dateutil\parser\_parser.py:1368, in parse(timestr, parserinfo, **kwargs)
   1366     return parser(parserinfo).parse(timestr, **kwargs)
   1367 else:
-> 1368     return DEFAULTPARSER.parse(timestr, **kwargs)

File ....envs\py310\lib\site-packages\dateutil\parser\_parser.py:651, in parser.parse(self, timestr, default, ignoretz, tzinfos, **kwargs)
    649     ret = self._build_naive(res, default)
    650 except ValueError as e:
--> 651     six.raise_from(ParserError(str(e) + ": %s", timestr), e)
    653 if not ignoretz:
    654     ret = self._build_tzaware(ret, res, tzinfos)

File :3, in raise_from(value, from_value)

ParserError: year 0 is out of range: 0000-00-00 00:00:00 present at position 0
```


### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

Hi @ededovic!

It seems like maybe there's both a mismatch in the issue description as well as some missing code that triggers the error (thanks for showing the error output!).

Using this code I can definitely **connect** to and **access** the table from ibis:

```python
sql = """
CREATE TEMPORARY TABLE ztmp_date_issue
(
    name      CHAR(10) NULL,
    tradedate DATETIME NOT NULL,
    date      DATETIME NULL
);
"""
with con.begin() as c:
    c.exec_driver_sql(sql)
    c.exec_driver_sql(
        """
        INSERT INTO ztmp_date_issue VALUES
            ('C', '2018-10-22', 0),
            ('B', '2017-06-07', 0),
            ('C', '2022-12-21', 0)
        """
    )
t = con.table("ztmp_date_issue")
```

You are correct that the issue does bubble up from pandas. This is visible when calling

```python
t.execute()
```

which shows this error:

```
E   pandas._libs.tslibs.parsing.DateParseError: year 0 is out of range: 0000-00-00 00:00:00, at position 0
```

--- a/ibis/backends/base/sql/alchemy/__init__.py
+++ b/ibis/backends/base/sql/alchemy/__init__.py
@@ -437,18 +437,23 @@ class BaseAlchemyBackend(BaseSQLBackend):
         else:
             util.log(query_str)
 
+    @staticmethod
+    def _new_sa_metadata():
+        return sa.MetaData()
+
     def _get_sqla_table(
         self, name: str, schema: str | None = None, autoload: bool = True, **_: Any
     ) -> sa.Table:
         # If the underlying table (or more likely, view) has changed, remove it
         # to ensure a correct reflection
+        meta = self._new_sa_metadata()
         with warnings.catch_warnings():
             warnings.filterwarnings(
                 "ignore", message="Did not recognize type", category=sa.exc.SAWarning
             )
             table = sa.Table(
                 name,
-                sa.MetaData(),
+                meta,
                 schema=schema,
                 autoload_with=self.con if autoload else None,
                 quote=self.compiler.translator_class._quote_table_names,

--- a/ibis/backends/mysql/__init__.py
+++ b/ibis/backends/mysql/__init__.py
@@ -7,11 +7,12 @@ import warnings
 from typing import Iterable, Literal
 
 import sqlalchemy as sa
+from sqlalchemy.dialects import mysql
 
 import ibis.expr.datatypes as dt
 from ibis.backends.base.sql.alchemy import BaseAlchemyBackend
 from ibis.backends.mysql.compiler import MySQLCompiler
-from ibis.backends.mysql.datatypes import _type_from_cursor_info
+from ibis.backends.mysql.datatypes import MySQLDateTime, _type_from_cursor_info
 
 
 class Backend(BaseAlchemyBackend):
@@ -112,6 +113,16 @@ class Backend(BaseAlchemyBackend):
                     warnings.warn("Unable to set session timezone to UTC.")
 
         super().do_connect(engine)
+
+    def _new_sa_metadata(self):
+        meta = super()._new_sa_metadata()
+
+        @sa.event.listens_for(meta, "column_reflect")
+        def column_reflect(inspector, table, column_info):
+            if isinstance(column_info["type"], mysql.DATETIME):
+                column_info["type"] = MySQLDateTime()
+
+        return meta
 
     def _metadata(self, query: str) -> Iterable[tuple[str, dt.DataType]]:
         if (

--- a/ibis/backends/mysql/__init__.py
+++ b/ibis/backends/mysql/__init__.py
@@ -114,8 +114,9 @@ class Backend(BaseAlchemyBackend):
 
         super().do_connect(engine)
 
-    def _new_sa_metadata(self):
-        meta = super()._new_sa_metadata()
+    @staticmethod
+    def _new_sa_metadata():
+        meta = sa.MetaData()
 
         @sa.event.listens_for(meta, "column_reflect")
         def column_reflect(inspector, table, column_info):

--- a/ibis/backends/mysql/datatypes.py
+++ b/ibis/backends/mysql/datatypes.py
@@ -7,6 +7,7 @@ from sqlalchemy.dialects import mysql
 from sqlalchemy.dialects.mysql.base import MySQLDialect
 
 import ibis.expr.datatypes as dt
+from ibis.backends.base.sql.alchemy import to_sqla_type
 
 # binary character set
 # used to distinguish blob binary vs blob text
@@ -253,3 +254,20 @@ def mysql_tinyint(satype, nullable=True):
 @dt.dtype.register(mysql.BLOB)
 def mysql_blob(satype, nullable=True):
     return dt.Binary(nullable=nullable)
+
+
+class MySQLDateTime(mysql.DATETIME):
+    """Custom DATETIME type for MySQL that handles zero values."""
+
+    def result_processor(self, *_):
+        return lambda v: None if v == "0000-00-00 00:00:00" else v
+
+
+@dt.dtype.register(MySQLDateTime)
+def mysql_timestamp(_, nullable=True):
+    return dt.Timestamp(nullable=nullable)
+
+
+@to_sqla_type.register(MySQLDialect, dt.Timestamp)
+def _mysql_timestamp(*_):
+    return MySQLDateTime()

--- a/ibis/backends/mysql/tests/test_client.py
+++ b/ibis/backends/mysql/tests/test_client.py
@@ -1,3 +1,7 @@
+from datetime import date
+
+import pandas as pd
+import pandas.testing as tm
 import pytest
 from pytest import param
 
@@ -93,3 +97,36 @@ def tmp_t(con_nodb):
 def test_get_schema_from_query_other_schema(con_nodb):
     t = con_nodb.table("t", schema="test_schema")
     assert t.schema() == ibis.schema({"x": dt.string})
+
+
+def test_zero_timestamp_data(con):
+    sql = """
+    CREATE TEMPORARY TABLE ztmp_date_issue
+    (
+        name      CHAR(10) NULL,
+        tradedate DATETIME NOT NULL,
+        date      DATETIME NULL
+    );
+    """
+    with con.begin() as c:
+        c.exec_driver_sql(sql)
+        c.exec_driver_sql(
+            """
+            INSERT INTO ztmp_date_issue VALUES
+                ('C', '2018-10-22', 0),
+                ('B', '2017-06-07', 0),
+                ('C', '2022-12-21', 0)
+            """
+        )
+    t = con.table("ztmp_date_issue")
+    result = t.execute()
+    expected = pd.DataFrame(
+        {
+            "name": ["C", "B", "C"],
+            "tradedate": pd.to_datetime(
+                [date(2018, 10, 22), date(2017, 6, 7), date(2022, 12, 21)]
+            ),
+            "date": [pd.NaT, pd.NaT, pd.NaT],
+        }
+    )
+    tm.assert_frame_equal(result, expected)

--- a/ibis/backends/sqlite/__init__.py
+++ b/ibis/backends/sqlite/__init__.py
@@ -65,7 +65,7 @@ class ISODATETIME(DATETIME):
     list of datetime formats SQLite accepts.
     """
 
-    def result_processor(self, value, dialect):
+    def result_processor(self, *_):
         return to_datetime
 
 
@@ -171,9 +171,8 @@ class Backend(BaseAlchemyBackend):
         with self.begin() as con:
             con.exec_driver_sql(f"ATTACH DATABASE {str(path)!r} AS {self._quote(name)}")
 
-    def _get_sqla_table(
-        self, name: str, schema: str | None = None, autoload: bool = True, **_: Any
-    ) -> sa.Table:
+    @staticmethod
+    def _new_sa_metadata():
         meta = sa.MetaData()
 
         @sa.event.listens_for(meta, "column_reflect")
@@ -181,12 +180,7 @@ class Backend(BaseAlchemyBackend):
             if type(column_info["type"]) is TIMESTAMP:
                 column_info["type"] = ISODATETIME()
 
-        return sa.Table(
-            name,
-            meta,
-            schema=schema or self.current_database,
-            autoload_with=self.con if autoload else None,
-        )
+        return meta
 
     def table(self, name: str, database: str | None = None, **_: Any) -> ir.Table:
         """Create a table expression from a table in the SQLite database.
@@ -226,6 +220,8 @@ class Backend(BaseAlchemyBackend):
         view = f"__ibis_sqlite_metadata{util.guid()}"
 
         with self.begin() as con:
+            if query in self.list_tables():
+                query = f"SELECT * FROM {query}"
             # create a view that should only be visible in this transaction
             con.exec_driver_sql(f"CREATE TEMPORARY VIEW {view} AS {query}")
 

--- a/ibis/backends/sqlite/tests/test_functions.py
+++ b/ibis/backends/sqlite/tests/test_functions.py
@@ -712,5 +712,5 @@ def test_count_on_order_by(con):
     sorted_table = t.order_by(sort_key)
     expr = sorted_table.count()
     result = str(expr.compile().compile(compile_kwargs={'literal_binds': True}))
-    expected = "SELECT count(*) AS count \nFROM main.batting AS t0"
+    expected = "SELECT count(*) AS count \nFROM batting AS t0"
     assert result == expected

--- a/ibis/backends/tests/snapshots/test_string/test_rlike/sqlite/out.sql
+++ b/ibis/backends/tests/snapshots/test_string/test_rlike/sqlite/out.sql
@@ -14,6 +14,6 @@ SELECT
   t0.timestamp_col,
   t0.year,
   t0.month
-FROM main.functional_alltypes AS t0
+FROM functional_alltypes AS t0
 WHERE
   _IBIS_SQLITE_REGEX_SEARCH(t0.string_col, '0')

